Source/AwsCommonRuntimeKit/mqtt/Mqtt5Types.swift

```diff
@@ -0,0 +1,795 @@
+    let wildcardSubscriptionsAvailable: Bool
+
+    /// Whether the server supports subscription identifiers
+    let subscriptionIdentifiersAvailable: Bool
+
+    /// Whether the server supports shared subscriptions
+    let sharedSubscriptionsAvailable: Bool
+
+    /// Whether the client has rejoined an existing session.
+    let rejoinedSession: Bool
+
+    /// The final client id in use by the newly-established connection.  This will be the configured client id if one was given in the configuration, otherwise, if no client id was specified, this will be the client id assigned by the server.  Reconnection attempts will always use the auto-assigned client id, allowing for auto-assigned session resumption.
+    let clientId: String
+
+    init (maximumQos: QoS, sessionExpiryIntervalSec: UInt32, receiveMaximumFromServer: UInt16, maximumPacketSizeToServer: UInt32,
+        topicAliasMaximumToServer: UInt16, topicAliasMaximumToClient: UInt16, serverKeepAliveSec: UInt16, retainAvailable: Bool,
+        wildcardSubscriptionsAvailable: Bool, subscriptionIdentifiersAvailable: Bool, sharedSubscriptionsAvailable: Bool, rejoinedSession: Bool,
+        clientId: String) {
+            self.maximumQos = maximumQos
+            self.sessionExpiryIntervalSec = sessionExpiryIntervalSec
+            self.receiveMaximumFromServer = receiveMaximumFromServer
+            self.maximumPacketSizeToServer = maximumPacketSizeToServer
+            self.topicAliasMaximumToServer = topicAliasMaximumToServer
+            self.topicAliasMaximumToClient = topicAliasMaximumToClient
+            self.serverKeepAliveSec = serverKeepAliveSec
+            self.retainAvailable = retainAvailable
+            self.wildcardSubscriptionsAvailable = wildcardSubscriptionsAvailable
+            self.subscriptionIdentifiersAvailable = subscriptionIdentifiersAvailable
+            self.sharedSubscriptionsAvailable = sharedSubscriptionsAvailable
+            self.rejoinedSession = rejoinedSession
+            self.clientId = clientId
+        }
+}
+
+/// Data model of an `MQTT5 CONNECT <https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901033>`_ packet.
+public class ConnectOptions {
+
+    /// The maximum time interval, in seconds, that is permitted to elapse between the point at which the client finishes transmitting one MQTT packet and the point it starts sending the next.  The client will use PINGREQ packets to maintain this property. If the responding CONNACK contains a keep alive property value, then that is the negotiated keep alive value. Otherwise, the keep alive sent by the client is the negotiated value.
+    var keepAliveIntervalSec: UInt16?
+
+    /// A unique string identifying the client to the server.  Used to restore session state between connections. If left empty, the broker will auto-assign a unique client id.  When reconnecting, the mqtt5 client will always use the auto-assigned client id.
+    var clientId: String?
```

Comment:
I remember IoT had a bug with auto-assigned client id. Should we force assign a id for now to work around it?

Reply:
I vaguely remember something like that but I also think if you don't provide a clientId in MQTT5, the broker assigns one to you and you get it back in the negotiated settings.